plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPropertyPage.java

```diff
@@ -202,7 +202,7 @@ private void evaluateFacetConfiguration() {
   private void updatePageBasedOnFacetConfig(boolean hasCorrectFacetConfiguration) {
     if (hasCorrectFacetConfiguration) {
       canSetMessage = true;
-      setErrorMessage(null);
+      content.getDataBindingContext().updateTargets();
```

Comment:
BTW, I remember https://github.com/GoogleCloudPlatform/google-cloud-eclipse/pull/880#issuecomment-256149028 was happening even though if I did not switch facets, so it was critical to fix it. Now it happens only after switching facets, so it is a slightly different bug.